src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -834,6 +818,20 @@ private bool InferConnectionDetails()
                     ReportSNIError(SNIProviders.INVALID_PROV);
                     return false;
                 }
+
+                try
```

Comment:
If `DataSource` should do nothing but parsing, the protocol falling back logic also need to be removed out of `DataSource`. Now it has a logic that tells the protocol is `TCP` if no protocol specified in connection string, and does not start with `\\` It is an additional calculation on top of parsing, which is the reason you want to remove SSRP out of  DataSource.